vpc/tool/container2/setup_container_linux.go

```diff
@@ -145,35 +145,73 @@ func addIPv4AddressAndRoute(ctx context.Context, nsHandle *netlink.Handle, link
 	}
 
 	gateway := cidr.Inc(ip.Mask(mask))
-	logger.G(ctx).WithField("gateway", gateway).Debug("Adding default route")
-	newRoute := netlink.Route{
-		Gw:        gateway.To4(),
-		Src:       ip,
-		LinkIndex: link.Attrs().Index,
-	}
-	if mtu != nil {
-		newRoute.MTU = *mtu
-	}
-	err = nsHandle.RouteAdd(&newRoute)
-	if err != nil {
-		logger.G(ctx).WithError(err).Error("Unable to add route to link")
-		return errors.Wrap(err, "Unable to add default route to link")
+	var defaultRouteFound bool
+	if len(routes) > 0 {
+		for _, route := range routes {
+			if route.Family != vpcapi.AssignIPResponseV3_Route_IPv4 {
+				continue
+			}
+			logger.G(ctx).WithField("route", route).Debug("Adding route")
+			_, routeNet, err := net.ParseCIDR(route.Destination)
+			if err != nil {
+				logger.G(ctx).WithError(err).Error("Could not parse route")
+				return fmt.Errorf("Could not parse route CIDR (%s): %w", route.Destination, err)
+			}
+			if routeNet.String() == "0.0.0.0/0" {
+				defaultRouteFound = true
+			}
+			newRoute := netlink.Route{
+				Gw:        gateway.To4(),
+				Src:       ip,
+				LinkIndex: link.Attrs().Index,
+				Dst:       routeNet,
+				MTU:       int(route.Mtu),
+			}
+			err = nsHandle.RouteAdd(&newRoute)
+			if err != nil {
+				logger.G(ctx).WithField("route", route).WithError(err).Error("Unable to add route to link")
+				return fmt.Errorf("Unable to add route %v to link due to: %w", route, err)
+			}
+		}
+		if !defaultRouteFound {
+			logger.G(ctx).Warn("Did not install default route. Suspicious.")
```

Comment:
Did not install ipv4 default route right?

Reply:
Yeah. We never install an IPv6 default route. We rely on SLAAC for that